fraudserv/service.go

```diff
@@ -181,6 +196,22 @@ func (f *ProofService) processIncoming(
 			"err", err, "proofType", proof.Type(), "height", proof.Height())
 		return pubsub.ValidationIgnore
 	}
+
+	// execute the verifier for proof type if exists
+	f.verifiersLk.RLock()
+	defer f.verifiersLk.RUnlock()
+	if verifier, ok := f.verifiers[proofType]; ok {
+		status, err := verifier(proof)
+		if err != nil {
+			log.Errorw("failed to run the verifier, err:", err, "proofType", proof.Type())
+			return pubsub.ValidationReject
+		}
+		if !status {
+			log.Errorw("failed to verify fraud proof for proofType", proof.Type())
+			return pubsub.ValidationReject
+		}
+	}
+
```

Comment:
```suggestion
	f.verifiersLk.RLock()
	verifier, ok := f.verifiers[proofType]
	f.verifiersLk.RUnlock()
	if ok {
		status, err := verifier(proof)
		if err != nil {
			log.Errorw("failed to run the verifier, err:", err, "proofType", proof.Type())
			return pubsub.ValidationReject
		}
		if !status {
			log.Errorw("failed to verify fraud proof for proofType", proof.Type())
			return pubsub.ValidationReject
		}
	}

```

Reply:
I'd suggest keeping it without if statement because otherwise verifierLK will be locked until we'll return from `processIncoming`